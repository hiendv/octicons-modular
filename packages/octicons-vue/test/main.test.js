import { mount } from '@vue/test-utils'
const { Octicon, zap } = require('../lib/main.js')

describe('Octicon component', () => {
  test('invalid', () => {
    const wrapper = mount(Octicon, {
      context: {
        props: {
          icon: {
            attrs () {},
            path () {}
          }
        }
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('valid', () => {
    const wrapper = mount(Octicon, {
      context: {
        props: { icon: zap }
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('scalable', () => {
    const wrapper = mount(Octicon, {
      context: {
        props: { icon: zap, scale: 2 }
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('scalable with 0', () => {
    const wrapper = mount(Octicon, {
      context: {
        props: { icon: zap, scale: 0 }
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('scalable with fractional values', () => {
    const wrapper = mount(Octicon, {
      context: {
        props: { icon: zap, scale: 1.2345 }
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('accepts an additional class', () => {
    const wrapper = mount(Octicon, {
      context: {
        props: { icon: zap, className: 'an-additional-class' }
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('ignores invalid additional classes', () => {
    const wrapper = mount(Octicon, {
      context: {
        props: { icon: zap, className: null }
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('accepts an label', () => {
    const wrapper = mount(Octicon, {
      context: {
        props: { icon: zap, label: 'label' }
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
