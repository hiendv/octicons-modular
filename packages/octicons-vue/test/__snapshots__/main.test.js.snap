// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Octicon component accepts an additional class 1`] = `"<svg version=\\"1.1\\" width=\\"10\\" height=\\"16\\" viewBox=\\"0 0 10 16\\" aria-hidden=\\"true\\" class=\\"octicon octicon-zap an-additional-class\\"><path fill-rule=\\"evenodd\\" d=\\"M10 7H6l3-7-9 9h4l-3 7 9-9z\\"></path></svg>"`;

exports[`Octicon component accepts an label 1`] = `"<svg version=\\"1.1\\" width=\\"10\\" height=\\"16\\" viewBox=\\"0 0 10 16\\" aria-label=\\"label\\" class=\\"octicon octicon-zap\\"><path fill-rule=\\"evenodd\\" d=\\"M10 7H6l3-7-9 9h4l-3 7 9-9z\\"></path></svg>"`;

exports[`Octicon component ignores invalid additional classes 1`] = `"<svg version=\\"1.1\\" width=\\"10\\" height=\\"16\\" viewBox=\\"0 0 10 16\\" aria-hidden=\\"true\\" class=\\"octicon octicon-zap\\"><path fill-rule=\\"evenodd\\" d=\\"M10 7H6l3-7-9 9h4l-3 7 9-9z\\"></path></svg>"`;

exports[`Octicon component scalable 1`] = `"<svg version=\\"1.1\\" width=\\"20\\" height=\\"32\\" viewBox=\\"0 0 10 16\\" aria-hidden=\\"true\\" class=\\"octicon octicon-zap\\"><path fill-rule=\\"evenodd\\" d=\\"M10 7H6l3-7-9 9h4l-3 7 9-9z\\"></path></svg>"`;

exports[`Octicon component scalable with 0 1`] = `"<svg version=\\"1.1\\" width=\\"0\\" height=\\"0\\" viewBox=\\"0 0 10 16\\" aria-hidden=\\"true\\" class=\\"octicon octicon-zap\\"><path fill-rule=\\"evenodd\\" d=\\"M10 7H6l3-7-9 9h4l-3 7 9-9z\\"></path></svg>"`;

exports[`Octicon component scalable with fractional values 1`] = `"<svg version=\\"1.1\\" width=\\"12.34\\" height=\\"19.75\\" viewBox=\\"0 0 10 16\\" aria-hidden=\\"true\\" class=\\"octicon octicon-zap\\"><path fill-rule=\\"evenodd\\" d=\\"M10 7H6l3-7-9 9h4l-3 7 9-9z\\"></path></svg>"`;

exports[`Octicon component valid 1`] = `"<svg version=\\"1.1\\" width=\\"10\\" height=\\"16\\" viewBox=\\"0 0 10 16\\" aria-hidden=\\"true\\" class=\\"octicon octicon-zap\\"><path fill-rule=\\"evenodd\\" d=\\"M10 7H6l3-7-9 9h4l-3 7 9-9z\\"></path></svg>"`;
